libraries/ts-command-line/src/CommandLineParameterProvider.ts

```diff
@@ -127,12 +144,35 @@ abstract class CommandLineParameterProvider {
     for (const parameter of this._parameters) {
       parameter._setValue(data);
     }
+
+    this._parameterMetadata.forEach((parameterMetadata: IParameterMetadata<any>) => { // tslint:disable-line:no-any
+      if (parameterMetadata.parameter.value === undefined && parameterMetadata.defaultValue !== undefined) {
+        parameterMetadata.parameter._setValue({
+          action: '',
+          [parameterMetadata.parameter._key]: parameterMetadata.defaultValue
+        });
+      }
+    });
+  }
+
+  protected validateParameters(): void {
+    const missingParameterLongNames: string[] = [];
+    // tslint:disable-next-line:no-any
+    this._parameterMetadata.forEach((parameterMetadata: IParameterMetadata<any>, parameterLongName: string) => {
+      if (parameterMetadata.parameter.value === undefined && parameterMetadata.required) {
+        missingParameterLongNames.push(parameterLongName);
+      }
+    });
+
+    if (missingParameterLongNames.length > 0) {
+      throw new Error(`Missing required parameters: ${missingParameterLongNames.join(', ')}`);
```

Comment:
>throw new Error(`Missing required parameters: ${missingParameterLongNames.join(', ')}`); [](start = 6, length = 88)

I thought "required" parameters were a feature of argparse -- we have to implement it ourself?